pkg/util/strategicpatch/patch.go

```diff
@@ -167,6 +213,12 @@ func mergeSlice(original, patch []interface{}, elemType reflect.Type, mergeKey s
 	if t.Kind() != reflect.Map {
 		// Maybe in the future add a "concat" mode that doesn't
 		// uniqify.
+		if !reflect.DeepEqual(original, patch) {
+			dw.WriteRemoval(fmt.Sprintf("%v", original))
+			dw.WriteAddition(fmt.Sprintf("%v", patch))
+		} else {
+			dw.WriteSame(fmt.Sprintf("%v", original))
+		}
```

Comment:
This looks wrong to me. List of scalars are supposed to be uniqified rather than replaced, i.e.
given original=  
- a
- b
- c
patch=
- a
- d
then merged=
- a
- b
- c
- d,
not just
- a
- d